test/parallel/test-console-async-write-error.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+require('../common');
+const { Console } = require('console');
+const { Writable } = require('stream');
+const assert = require('assert');
+
+const out = new Writable({
+  write(chunk, enc, callback) {
```

Comment:
Just to be on the safe side, there should likely be a check that verifies this version of `write` is being called. I'm not certain if `common.mustCall()` would work here.